chan.h

```diff
@@ -12,43 +12,59 @@ using std::queue;
 using std::unique_lock;
 
 template <typename T> class Chan {
+  // buffered queue properties
   queue<T> que;
   mutable mutex mtx;
-  condition_variable cond;
+  condition_variable r_cond;
+  condition_variable w_cond;
 
-  const size_t quota;
+  int r_waiting;
+  int w_waiting;
```

Comment:
as far as I know, variables in structs and classes in C++ won't initialize to zero by default. 
you should initialize them in constructor beforehand.

Reply:
I did.